python/cudf/cudf/core/column/column.py

```diff
@@ -1317,6 +1317,15 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
 
     elif np.isscalar(arbitrary) and not isinstance(arbitrary, memoryview):
         length = length or 1
+        if (
+            (nan_as_null is True)
+            and not isinstance(arbitrary, str)
```

Comment:
Actually I think we can do this:
```suggestion
            and isinstance(arbitrary,(np.floating, float))
```

Reply:
That should handle both numpy and python floats cleanly so we don't waste time checking integers unecessarily.